### PR TITLE
feat: add types for multi-routing api

### PR DESCRIPTION
**Pull request by Ikari-Shinji-re**

### CODE REVIEW

File: src/api/main/routing.ts [hunk 1]
```diff
@@ -173,4 +173,110 @@ export type BestRouteResponse = {
...
+  results: MultiRouteSimulationResult[]
+}
+
+/**
+ * The body of confirmation request for selected route.
+ * @property {string} requestId - The unique requestId which is generated for this request by the server. It should be
+ * passed down to all other endpoints if this swap continues on. e.g. d10657ce-b13a-405c-825b-b47f8a5016ad * @property {{ [key: string]: string }} selectedWallets - Map of blockchain to selected address
+ * @property {string} [destination] - Custom destination for the route
+ */
+export type ConfirmSwapRequest = {
```
Comment (RanGojo) [inline src/api/main/routing.ts:264]: This is my suggestion for types and method names.

```ts
getBestRoute(request: BestRouteRequest) -> BestRouteResponse
getAllRoutes(request: MultiRouteRequest) -> MultiRouteResponse // or maybe getMultiRoutes
confirmMultiRouteRequest(request: ConfirmMultiRouteRequest) -> ConfirmMultiRouteResponse
```

File: src/api/main/routing.ts [hunk 2]
```diff
@@ -173,4 +173,110 @@ export type BestRouteResponse = {
...
+export type BestsRoutesRequest = BestRouteRequest
+
+/**
+ * The best route response for multi-routing, if the result fields is null, it means that no route is found
+ *
+ * @property {MultiRouteSimulationResult} results - List of best routes data
+ */
+export type BestsRouteResponse = BestRouteResponse & {
+  results: MultiRouteSimulationResult[]
+}
```
Comment (RanGojo) [inline src/api/main/routing.ts:256]: I think we have some partial fields from `BestRouteResponse` not all of them in multi route response. (for example we don't have fields related to validation afaik) also you have missed some fields like `routeId`

File: src/api/main/routing.ts [hunk 3]
```diff
@@ -173,4 +173,110 @@ export type BestRouteResponse = {
...
+ * @property {RoutingResultType} resultType
+ * @property {SwapResult[]} swaps - List of required swaps to swap X to Y with the expected outputAmount
+ * @property {SimulationScore[]} scores
+ * @property {RouteTag[]} tags
+ * @property {BlockchainValidationStatus[] | null} validationStatus - Pre-requisites check result. It will be null if
+ * the request checkPrerequisites was false
+ * @property {string[]} missingBlockchains - List of all blockchains which are necessary to be present for the best
+ * route and user has not provided any connected wallets for it. A null or empty list indicates that there is no problem.
+ * @property {boolean} walletNotSupportingFromBlockchain - A warning indicates that none of your wallets have the same
+ * blockchain as X asset
```
Comment (RanGojo) [inline src/api/main/routing.ts:230]: Please complete doc for missing parts and double check it.

**Review by RanGojo — CHANGES_REQUESTED** [review]

File: src/api/main/routing.ts (hunk 2 above)
Comment (Ikari-Shinji-re) [reply src/api/main/routing.ts:256]: I included the routeId field in the MultiRouteResponse and verified the types. All remaining fields are identical between the BestRouteResponse and MultiRouteResponse.

File: src/api/main/routing.ts [hunk 4]
```diff
@@ -173,4 +174,116 @@ export type BestRouteResponse = {
...
+/**
+ * The best route response for multi-routing, if the result fields is null, it means that no route is found
+ *
+ * @property {string} routeId - The unique roteId generated for this request by server
+ * @property {MultiRouteSimulationResult} results - List of best routes data
+ */
+export type MultiRouteResponse = BestRouteResponse & {
+  routeId: string
+  results: MultiRouteSimulationResult[]
+}
```
Comment (RanGojo) [inline src/api/main/routing.ts:260]: You need to exclude some fields like request id from BestRouteResponse.

File: src/api/main/routing.ts [hunk 5]
```diff
@@ -173,4 +174,116 @@ export type BestRouteResponse = {
...
+
+/**
+ * The response of confirmation request for selected route
+ *
+ * @property {string | null} error - Error message about the incident if ok == false.
+ * @property {string | null} errorCode - Error code about the incident if ok == false.
+ * @property {boolean} ok - If true, the result has a value and error message is null.
+ * @property {BestRouteResponse | null} result - result of confirm swap.
+ */
+export type ConfirmSwapResponse = {
```
Comment (RanGojo) [inline src/api/main/routing.ts:284]: I think `ConfirmRouteRequest` and `ConfirmRouteResponse` is better as swap word is used for  our basic api.

File: src/api/main/routing.ts [hunk 6]
```diff
@@ -173,4 +174,116 @@ export type BestRouteResponse = {
...
+  error: string | null
+  errorCode: string | null
+  ok: boolean
+  result: BestRouteResponse | null
```
Comment (RanGojo) [inline src/api/main/routing.ts:288]: Are you sure if type of `result` is correct? We don't have for example tag and scores in confirm response? 🤔

**Review by RanGojo — CHANGES_REQUESTED** [review]

**Review by RanGojo — APPROVED** [review]

**Pull request merged**